carma_wm_ctrl/src/WMBroadcaster.cpp

```diff
@@ -369,6 +374,115 @@ void WMBroadcaster::removeGeofence(std::shared_ptr<Geofence> gf_ptr)
+
+  exit(0);
+}
+
+  for(auto id : route_msg.route_path_lanelet_ids) 
+  {
+    auto laneLayer = current_map_->laneletLayer.get(id);
+    path.push_back(laneLayer);
+  }
+ 
+  
+  if(path.size() == 0) exit(0);
+  ROS_WARN_STREAM("Got here safely");
+   /*logic to determine route bounds*/
+  std::vector<lanelet::ConstLanelet> llt; 
+  std::vector<lanelet::BoundingBox3d> pathBox; 
+  float minX = 99999;
+  float minY = 99999;
+  float minZ = 99999;
+  float maxX = -99999;
+  float maxY = -99999;
+  float maxZ = -99999;
+  ROS_WARN_STREAM("Got to this point safely");
+
+  while (path.size() != 0) //Continue until there are no more lanelet elements in path
+  {
+      llt.push_back(path.back()); //Add a lanelet to the vector
+    
+
+      pathBox.push_back(lanelet::geometry::boundingBox3d(llt.back())); //Create a bounding box of the added lanelet and add it to the vector
+
+
+      if (pathBox.back().corner(lanelet::BoundingBox3d::BottomLeft).x() < minX)
+        minX = pathBox.back().corner(lanelet::BoundingBox3d::BottomLeft).x(); //minimum x-value
+
+
+      if (pathBox.back().corner(lanelet::BoundingBox3d::BottomLeft).y() < minY)
+        minY = pathBox.back().corner(lanelet::BoundingBox3d::BottomLeft).y(); //minimum y-value
+
+    if (pathBox.back().corner(lanelet::BoundingBox3d::BottomLeft).z() < minZ)
+        minZ = pathBox.back().corner(lanelet::BoundingBox3d::BottomLeft).z(); //minimum z-value
+
+
+     if (pathBox.back().corner(lanelet::BoundingBox3d::TopRight).x() > maxX)
+        maxX = pathBox.back().corner(lanelet::BoundingBox3d::TopRight).x(); //maximum x-value
+
+
+        if (pathBox.back().corner(lanelet::BoundingBox3d::TopRight).y() > maxY)
+        maxY = pathBox.back().corner(lanelet::BoundingBox3d::TopRight).y(); //maximum y-value
+
+     if (pathBox.back().corner(lanelet::BoundingBox3d::TopRight).z() > maxZ)
+        maxZ = pathBox.back().corner(lanelet::BoundingBox3d::TopRight).z(); //maximum z-value
+
+
+      path.pop_back(); //remove the added lanelet from path an reduce pack.size() by 1
+  } //end of while loop
+
+  ROS_WARN_STREAM("Successfully exited");
+
+  
+
+  std::string target_frame = base_map_georef_;
+  lanelet::projection::LocalFrameProjector local_projector(target_frame.c_str());
+  lanelet::BasicPoint3d localRoute;
+
+  localRoute.x()= minX;
+  localRoute.y()= minY;
+  localRoute.z()= minZ; 
+
+  lanelet::GPSPoint gpsRoute = local_projector.reverse(localRoute); //If the appropriate library is included, the reverse() function can be used to convert from local xyz to lat/lon
+
+  cav_msgs::ControlRequest cR; /*Fill the latitude value in message cB with the value of lat */
+  cav_msgs::ControlBounds cB; /*Fill the longitude value in message cB with the value of lon*/
+  cB.latitude = gpsRoute.lat;
+  cB.longitude = gpsRoute.lon;
+
+  cB.offsets[0] = maxX - minX;
+  cB.offsets[1] = maxY - minY;
+  cB.offsets[2] = maxZ - minZ;
+
```

Comment:
Add a TODO comment here to update this for new geofence spec. The spec you had for offsets was old, but we can update once the message updates are completed.